### DISCUSSION TITLE
Worksheet "Header & Footer" Text

### DISCUSSION THREAD

**Discussion by anthonybarbatto** [exceljs/exceljs#2523] [Ideas & Proposals]
I have a worksheet that includes data in the header & footer (not column headers) and I'm wondering if this text is accessible via the worksheet object? I've been looking through the code but haven't seemed to find it.

In the screenshot below, I'm interested in getting the Month, Year text.

Thank you!

![2019 2019-01-21 13-09-45](https://user-images.githubusercontent.com/744609/51499206-32fc6f80-1d7e-11e9-9964-57152bd164b5.jpg)

#helpwanted

**Comment by Siemienik** [comment]
@anthonybarbatto I have the same issue, as far as i know, it looks like this feature is not implemented yet

**Comment by donaldsonjulia** [comment]
@Siemienik I had implemented a really quick fix on a forked branch for getting a header and footer into an app I needed it for. I'm currently working on creating a proper PR for header/footer to make it align with the desired API as written in `TODO.txt`

**Comment by Siemienik** [comment]
@donaldsonjulia have you already created PR with fix?

**Comment by donaldsonjulia** [comment]
@Siemienik I've been out of town so now I'm returning to getting it cleaned up to create a PR

**Comment by donaldsonjulia** [comment]
@Siemienik I'm stuck with an issue where my attempt to include `evenHeader` seems to corrupt the entire file - `evenFooter` and `oddFooter` work, but as soon as I include `evenHeader` it's corrupted.....is this not valid xml?? My knowledge of OpenXML is limited. I was following the specs in `TODO.txt`:
```
<headerFooter differentOddEven="1">
  <oddHeader>&amp;Ltesting&amp;COdd Header&amp;Rtesting</oddHeader>
  <evenHeader>&amp;Ltesting&amp;CEven Header Test&amp;Rtesting</evenHeader>
  <oddFooter>&amp;LPrepared by Adrien Buckridge&amp;C&amp;P&amp;RTesting Inc</oddFooter>
  <evenFooter>&amp;Ltesting&amp;Ctesting even&amp;Rtesting</evenFooter>
</headerFooter>
```
I want to get this PR in but this is an unexpected hiccup that I just can't figure out and I'm trying to do it outside of work hours so its taking longer than I thought -- are you able to possibly help me with what might be the problem?
